internal/connectivitystate/connectivitystate.go

```diff
@@ -0,0 +1,93 @@
+/*
+ *
+ * Copyright 2023 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+// Package connectivitystate provides functionality to report and track
+// connectivity state changes of ClientConns and SubConns.
+package connectivitystate
+
+import (
+	"context"
+	"sync"
+
+	"google.golang.org/grpc/connectivity"
+	"google.golang.org/grpc/internal/callbackserializer"
+)
+
+// Watcher defines the functions which is executed as soon as a connectivity state changes
+// of Tracker is reported.
+type Watcher interface {
+	// OnStateChange is invoked when connectivity state changes on ClientConn is reported.
+	OnStateChange(state connectivity.State)
+}
+
+// Tracker manages watchers and their status. It holds a previous connecitivity state of
+// ClientConns and SubConns.
+type Tracker struct {
+	mu       sync.Mutex
+	state    connectivity.State
+	watchers map[Watcher]bool
+	cs       *callbackserializer.CallbackSerializer
+}
+
+// NewTracker returns a new Tracker instance.
+func NewTracker(state connectivity.State) *Tracker {
+	ctx := context.Background()
```

Comment:
What is the use of this context being passed to the `CallbackSerializer` without having a way to cancel it (which is when the goroutine spawned by the serializer will exit). 

So, we need to do the following here:
- Create this context with a cancel func. `ctx, cancel := context.WithCancel(context.Background()`
- Store the cancel func in a field inside the `Tracker`
- Add a `Stop()` method to the `Tracker` which calls this cancel func, thereby ensuring that resources allocated by the callback serializer are released.
- Ensure that the ClientConn calls `Stop()` when it is closed.

Reply:
It would be better. Thanks you for your advice!